pydantic/experimental/pipeline.py

```diff
@@ -21,18 +21,18 @@
 
     from pydantic import GetCoreSchemaHandler
 
+from pydantic._internal._internal_dataclass import frozen_true as _frozen_true
 from pydantic._internal._internal_dataclass import slots_true as _slots_true
 
 if sys.version_info < (3, 10):
-    # TODO: remove this once we drop 3.9
     EllipsisType = type(Ellipsis)
 else:
     from types import EllipsisType
 
 __all__ = ['validate_as', 'validate_as_deferred', 'transform']
 
 
-@dataclass(**_slots_true)
+@dataclass(**_slots_true, **_frozen_true)
```

Comment:
I would have suggested just adding a constant to this module rather than a new constant in `_internal_dataclass`, I was thinking:

```suggestion
_slots_frozen = {**_slots_true, 'frozen': True}

@dataclass(**_slots_frozen)
```
or similar, and repeated where it makes sense.

However, I'll also note that there doesn't seem to be much downside to just doing `frozen=True` instead of `**_frozen_true`; the reason we need `**_slots_true` is for it to work with versions of python that didn't have the slots kwarg for dataclasses, but I guess all of our supported versions of python have `frozen` as a supported kwarg.

Given you've made the change already I don't mind keeping it as-is if you don't want to change it again but I'm also okay with either of the above versions.

Reply:
Yeah makes sense, I'll go with `_slots_frozen`!

Reply:
Thanks for the feedback! Just pushed the change.